# How to use the MCMC output file

In [6]:
%matplotlib inline
import numpy as np
import TOFPlanet
import ppbs
import observables
import ahelpers as ah
import samplooker as spl

When the MCMC driver finished its job, it saves the resulting chain-ensemble in the file `final.npz` in the run directory. The periodic checkpoints in `dump0.npz` and `dump1.npz` have the same format. Let's look at an example.

In [7]:
fname = 'final.npz'
C, L = ah.load_chain(fname)
print(f'{C.shape=}')
print(f'{L.shape=}')

C.shape=(160, 4400, 8)
L.shape=(160, 4400)


The array `C`, for "chains," has all the numbers you need to define the planet models. Its shape is `w`, number of "walkers", by `n`, number of steps, by `d`, number of dimensions. For example this file is the output of a `drive-ppbs.py` run. The ppbs parameerization has 8 dimensions (plus potentially rotation period, which in this run was kept constant). This was a run of 160 walkers, taking 4400 steps each. A vector of parameters for one planet will be something like, say, `C[2,24,:]`, meaning the 25th step of the third walker. The walkers are a feature of the so-called "ensemble" MCMC algorithm. We need the walker-by-walker data only to diagnose sampling problems and figure out auto-correlation distance. Afterwards we will forget about walkers and end up with an array of something-by-8, of (hopefully) independent samples.

The array `L`, for "likelihood," has the likelihood values for each step of each walker. This is sometimes needed for debugging and diagnosing issues, or for detecting outliers.